From 47027e41ed04d09d8b2ad4bb66fd57b1414d0bcc Mon Sep 17 00:00:00 2001
From: Chunfeng Yun <chunfeng.yun@mediatek.com>
Date: Tue, 7 Jul 2015 17:16:20 +0800
Subject: [PATCH 2/6] dt-bindings: Add a binding for Mediatek xHCI host
 controller

add a DT binding documentation of xHCI host controller for the
MT8173 SoC from Mediatek.

Change-Id: I0ac0884c4bc6e3783c4f9cc028c061b019dd2825
Signed-off-by: Chunfeng Yun <chunfeng.yun@mediatek.com>
---
 .../devicetree/bindings/usb/mt8173-xhci.txt        | 27 ++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/usb/mt8173-xhci.txt

diff --git a/Documentation/devicetree/bindings/usb/mt8173-xhci.txt b/Documentation/devicetree/bindings/usb/mt8173-xhci.txt
new file mode 100644
index 0000000..94cec94
--- /dev/null
+++ b/Documentation/devicetree/bindings/usb/mt8173-xhci.txt
@@ -0,0 +1,27 @@
+MT65XX xhci
+
+The device node for Mediatek SOC usb3.0 host controller
+
+Required properties:
+ - compatible : supports "mediatek,mt8173-xhci"
+ - reg        : Offset and length of registers
+ - interrupts : Interrupt mode, number and trigger mode
+ - reg-vusb33-supply:  regulator of usb avdd3.3v
+ - reg-vbus-supply : regulator of vbus;
+ - usb-phy : the phy that xhci will bind
+ - usb3-lpm-capable: supports USB3 LPM
+ - clocks     : must support all clocks that xhci needs
+ - clock-names: should be "sys_mac" for sys and mac clocks
+
+Example:
+usb: usb30@11270000 {
+	compatible = "mediatek,mt8173-xhci";
+	reg = <0 0x11270000 0 0x1000>;
+	interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_LOW>;
+	reg-vusb33-supply = <&mt6397_vusb_reg>;
+	reg-vbus-supply = <&usb_p1_vbus>;
+	usb-phy = <&u3phy>;
+	usb3-lpm-capable;
+	clocks = <&topckgen CLK_TOP_USB30_SEL>;
+	clock-names = "sys_mac";
+};
-- 
1.9.1

